File: include/extras-knotifications/libknotificationpermission.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const QObject = @import("libqt6").QObject;
const qnamespace_enums = @import("../libqnamespace.zig").enums;

/// ### [Upstream resources](https://api.kde.org/knotificationpermission.html)
pub const KNotificationPermission = extern struct {
    /// ### [Upstream resources](https://api.kde.org/knotificationpermission.html)
    ///
    /// The pointer to the underlying Qt C++ object
    ///
    ptr: QtC.KNotificationPermission,

    pub const _is_KNotificationPermission = {};

    /// ### [Upstream resources](https://api.kde.org/knotificationpermission.html#checkPermission)
    ///
    /// ## Returns:
    ///
    /// ` qnamespace_enums.PermissionStatus `
    ///
    pub fn CheckPermission() i32 {
        return qtc.KNotificationPermission_CheckPermission();
    }

    /// ### [Upstream resources](https://api.kde.org/knotificationpermission.html#requestPermission)
    ///
    /// ## Parameter(s):
    ///
    /// ` context: QObject `
    ///
    /// ` callback: *const fn (funcparam1: qnamespace_enums.PermissionStatus) callconv(.c) void `
    ///
    pub fn RequestPermission(context: anytype, callback: *const fn (i32) callconv(.c) void) void {
        comptime _ = @TypeOf(context)._is_QObject;
        qtc.KNotificationPermission_RequestPermission(@ptrCast(context.ptr), @bitCast(@intFromPtr(callback)));
    }
};

File: src/extras-knotifications/libknotificationpermission.cpp
#include <KNotificationPermission>
#include <QObject>
#include <knotificationpermission.h>
#include "libknotificationpermission.h"
#include "libknotificationpermission.hxx"

int KNotificationPermission_CheckPermission() {
    return static_cast<int>(KNotificationPermission::checkPermission());
}

void KNotificationPermission_RequestPermission(QObject* context, intptr_t callback) {
    auto callback_func = [callback](Qt::PermissionStatus funcparam1_fp) -> void {
        int funcparam1_fv = static_cast<int>(funcparam1_fp);
        reinterpret_cast<void (*)(int)>(callback)(funcparam1_fv);
    };
    KNotificationPermission::requestPermission(context, callback_func);
}
